test/e2e/mnist_rayjob_raycluster_test.go

```diff
@@ -388,10 +401,42 @@ func constructRayCluster(_ Test, namespace *corev1.Namespace, mnist *corev1.Conf
 			},
 		},
 	}
+	if accelerator.IsGpu() {
+		raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Requests[corev1.ResourceName("nvidia.com/gpu")] = resource.MustParse("1")
+		raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Limits[corev1.ResourceName("nvidia.com/gpu")] = resource.MustParse("1")
+
+		if accelerator.ResourceLabel == "amd.com/gpu" {
+			// Remove the nvidia.com/gpu resource
+			delete(raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Requests, corev1.ResourceName("nvidia.com/gpu"))
+			delete(raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Limits, corev1.ResourceName("nvidia.com/gpu"))
+
+			// update with amd.com/gpu resource
+			raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Tolerations[0].Key = "amd.com/gpu"
+			raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Requests[corev1.ResourceName("amd.com/gpu")] = resource.MustParse("1")
+			raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Limits[corev1.ResourceName("amd.com/gpu")] = resource.MustParse("1")
+		}
+	}
```

Comment:
Can you rather remove `nvidia.com/gpu` from the "base" Ray cluster resources declaration above and just append it here?
That way you can avoid declaring explicit `nvidia.com/gpu` or `amd.com/gpu` and you will have all the GPU related changes in one place.